### PR TITLE
Release SNAPSHOT versions

### PR DESCRIPTION
**Pull request by brfrn169**
This PR adds a GitHub actions workflow to release SNAPSHOT versions. Please take a look!

### CODE REVIEW

File: .github/workflows/release-snapshot.yaml [hunk 1]
```diff
@@ -0,0 +1,54 @@
+name: Release SHAPSHOT
+
+on:
+  push:
+    branches:
+      - master
+      - "[0-9]+"
+      - "[0-9]+.[0-9]+"
```
Comment (brfrn169) [inline .github/workflows/release-snapshot.yaml:8]: This workflow targets the `master` branch, the support branches (like `3`), and the release branches (like `3.4` and `3.5`). The triggers are `push` to those branches.

File: .github/workflows/release-snapshot.yaml [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+            ${{ runner.os }}-gradle-
+
+      - name: Set version
+        id: version
+        run: |
+          VERSION=$(gradle :core:properties -q | grep "version:" | awk '{print $2}')
+          echo ::set-output name=version::${VERSION}
+
+      - name: Upload SNAPSHOT versions for scalardb, scalardb-server, scalardb-rpc, and scalardb-schema-loader to Maven Central Repository
+        if: contains(steps.version.outputs.version, '-SNAPSHOT')
```
Comment (brfrn169) [inline .github/workflows/release-snapshot.yaml:39]: It uploads the SNAPSHOT versions only when the project version contains `-SNAPSHOT`.

**Review by Torch3333 — APPROVED** [review]
LGTM, thank you!

File: .github/workflows/release.yaml [hunk 3]
```diff
@@ -94,10 +94,3 @@ jobs:
           asset_path: schema-loader/build/libs/scalardb-schema-loader-${{ steps.version.outputs.version }}.jar
           asset_name: scalardb-schema-loader-${{ steps.version.outputs.version }}.jar
           asset_content_type: application/java-archive
-
-      - name: Cleanup Gradle cache
```
Comment (feeblefakie) [inline .github/workflows/release.yaml:98]: OK to delete them? (I think it's auto-generated but not really sure why they are auto-generated.)

**Review by feeblefakie — APPROVED** [review]
LGTM! Thank you!

File: .github/workflows/release.yaml (hunk 3 above)
Comment (brfrn169) [reply .github/workflows/release.yaml:98]: Actually, I'm also not sure they are needed, so I deleted it. I will restore it if we have any problems with it. Thank you.

**Pull request merged**